chainer/train_imagenet.py

```diff
@@ -64,45 +91,38 @@ def train_loop():
     label = np.ndarray((args.batchsize), dtype=np.int32)
     label.fill(1)
     count = 0
+    timer = Timer()
     for i in range(niter):
         x = xp.asarray(data)
         y = xp.asarray(label)
         
         if args.arch == 'googlenet':
-            start = xp.cuda.Event()
-            end = xp.cuda.Event()
-            start.record()
+            timer.preprocess()
```

Comment:
Thanks, the reason we want to keep in the same script is because if there is any change in GPU benchmark, then it will do in CPU benchmark. And  I don't understand why it is not correct to measure the elapsed time on GPU, When xp==cupy, logical seems keep the same with original script, that is also what chainer hook function (in chainer/function_hooks/timer.py)did. Thanks very much

+class Timer():
 +    def preprocess(self):
 +        if xp == np:
 +            self.start = time.time()
 +        else:
 +            self.start = xp.cuda.Event()
 +            self.end = xp.cuda.Event()
 +            self.start.record()
 +    def postprocess(self):
 +        if xp == np:
 +            self.end = time.time()
 +        else:
 +            self.end.record()
 +            self.end.synchronize()
 +    def getElapseTime(self):
 +        if xp == np:
 +            return (self.end - self.start)*1000
 +        else:
 +            return xp.cuda.get_elapsed_time(self.start, self.end)